src/main/java/com/puppycrawl/tools/checkstyle/api/DetailAST.java

```diff
@@ -19,6 +19,8 @@
 
 package com.puppycrawl.tools.checkstyle.api;
 
+import org.checkerframework.checker.index.qual.NonNegative;
```

Comment:
this is apart I am not sure about Checker: we looks like have to change our api and force all our user to depend on Checker, that is not good.
We recently resolved big pain of dependency to guava, that is not api based dependency.
Do we have any other way resolve this ?
if not I would rather disable validation of Checker for cases that require `NonNegative`.
Can we remove it from `api` class only ?

Reply:
@romani ,
Sorry to involve.. But why are we not using  **jsr-305** annotations here ?
I think we have a Nonnegative annotation in javax. 
https://www.javadoc.io/doc/com.google.code.findbugs/jsr305/latest/javax/annotation/Nonnegative.html

@Lmh-java , could you please explain why you have introduced checker's annotations here? 
I understand why there are no ci troubles here. Checker framework treats annotations from other tools as if they were extracted from corresponding annotation in Checker itself. 
But by doing this, `we will only be increasing our JAR.`

Reply:
checkerframework does not support `Nonnegative` from javax. I was trying to use it first until I figured out it does not recognize that annotation. Therefore, I turned to use `NonNegative` from the checker framework.

Reply:
Yes, looks that way , I was just browsing through checker release docs : https://checkerframework.org/releases/2.3.1/api/index.html?javax/annotation/package-summary.html

> Do we have any other way resolve this ?

@romani , I'm afraid we don't have any alternative to migrate **jsr 305** other than checker framework.
Most of the projects are removing their javax annotations rather than using checker imports.
Jenkins for example : https://groups.google.com/g/jenkinsci-dev/c/uE1wwtVi1W0

Reply:
Yea, same here. I've just looked at Jenkin's discussion posts, but I will try to solve this problem by throwing an `UncheckException`. Not very sure if that works, but I will test it out.

Reply:
>It seems checkerframework does not support Nonnegative from javax.

Can we check if there is issue registed for it on checker ? If not, let's create.
If there is defect in checker, I would rather postpone dealing with such violations. There are tons of others, no need to be stuck with problematic.

Reply:
The Nonnegative annotation is just one of many..
These are all the annotations in package javax.annotations (jsr 305) : 
https://www.javadoc.io/doc/com.google.code.findbugs/jsr305/latest/javax/annotation/package-summary.html

And checker is only supporting three of them : 
https://checkerframework.org/releases/2.3.1/api/index.html?javax/annotation/package-summary.html

Reply:
> And checker is only supporting three of them   

Yea, I have seen that as well. 

> Can we check if there is issue registed for it on checker ? If not, let's create.

I have opened an issue to their repo already and I am waiting for their response. Meanwhile, I will be dealing with other violations. We can postpone this violation for now.

Reply:
I can solve this problem by introducing a throw of `IllegalArgumentException` as an unchecked exception. But I still feel it's not elegant and not necessary,